native/core/src/execution/datafusion/planner.rs

```diff
@@ -950,17 +962,29 @@ impl PhysicalPlanner {
                     //
                     // Note that `result_exprs` should only be set for final aggregation on the
                     // Spark side.
-                    Arc::new(ProjectionExec::try_new(result_exprs?, aggregate)?)
-                };
-
-                Ok((scans, exec))
+                    let projection =
+                        Arc::new(ProjectionExec::try_new(result_exprs?, aggregate.clone())?);
+                    Ok((
+                        scans,
+                        Arc::new(SparkPlan::new_with_additional(
+                            spark_plan.plan_id,
+                            projection,
+                            vec![child],
+                            vec![aggregate],
+                        )),
```

Comment:
This is an example where we are currently dropping the aggregate metrics and just capturing the project metrics